WordPress/Classes/ViewRelated/NUX/SignupGoogleViewController.swift

```diff
@@ -25,80 +25,129 @@ class SignupGoogleViewController: LoginViewController {
 
     override func viewDidAppear(_ animated: Bool) {
         super.viewDidAppear(animated)
-        if !hasShownGoogle {
-            showGoogleScreen()
-            hasShownGoogle = true
+        displayGoogleSingleSignOnIfNeeded()
+    }
+
+    private func displayGoogleSingleSignOnIfNeeded() {
+        guard !hasShownGoogle else {
+            return
         }
+
+        displayGoogleSingleSignOn()
+        hasShownGoogle = true
     }
 
-    private func showGoogleScreen() {
+    private func displayGoogleSingleSignOn() {
         GIDSignIn.sharedInstance().disconnect()
 
         // Flag this as a social sign in.
-        loginFields.meta.socialService = SocialServiceName.google
+        loginFields.meta.socialService = .google
 
         // Configure all the things and sign in.
-        GIDSignIn.sharedInstance().delegate = self
-        GIDSignIn.sharedInstance().uiDelegate = self
-        GIDSignIn.sharedInstance().clientID = WordPressAuthenticator.shared.configuration.googleLoginClientId
-        GIDSignIn.sharedInstance().serverClientID = WordPressAuthenticator.shared.configuration.googleLoginServerClientId
+        guard let googleSSO = GIDSignIn.sharedInstance() else {
```

Comment:
I don't think this bit is necessary for decoupling. is it?

Reply:
Not really needed, but we're saving 5 method calls, and replacing them with a single local reference